### PR TITLE
Fix issue with leaf/nodes in RecursiveXYCut that only have whitespaces words.

### PR DESCRIPTION
**Pull request by huzhiguan**
In VerticalCut and HorizontalCut, it's using `CountWords() <= 1` as part of the condition to stop cutting with small text segments. This function doesn't exclude all the whitespace words. I made changes to make sure that those two functions will return when finding  XYLeaf/XYNode only have whitespaces words.

In `GetBlocks()`, if the returned node is an XYLeaf node, the `node.GetLeafs()` will return null which will cause the following lines to throw an exception.  I modified that part to make sure it's handling leaf node differently.

### CODE REVIEW

File: src/UglyToad.PdfPig/DocumentLayoutAnalysis/RecursiveXYCut.cs [hunk 1]
```diff
@@ -84,6 +91,18 @@ public IReadOnlyList<TextBlock> GetBlocks(IEnumerable<Word> pageWords, decimal m
...
+
+            if(!words.Any())
+            {
+                return new XYNode(null);
+            }
+            else
+            {
+                //Create new leaf with non-whitespace words.
+                leaf = new XYLeaf(words);
+            }
```
Comment (BobLd) [inline src/UglyToad.PdfPig/DocumentLayoutAnalysis/RecursiveXYCut.cs]: Go to line here

File: src/UglyToad.PdfPig/DocumentLayoutAnalysis/RecursiveXYCut.cs [hunk 2]
```diff
@@ -177,14 +193,25 @@ public IReadOnlyList<TextBlock> GetBlocks(IEnumerable<Word> pageWords, decimal m
...
+
             if (leaf.CountWords() <= 1)
             {
                 // we stop cutting if 
                 // - only one word remains
                 return leaf;
             }
 
-            var words = leaf.Words.Where(w => !string.IsNullOrWhiteSpace(w.Text)).OrderBy(w => w.BoundingBox.Bottom).ToArray(); // order bottom to top
```
Comment (BobLd) [inline src/UglyToad.PdfPig/DocumentLayoutAnalysis/RecursiveXYCut.cs]: Remove line here

**Review by BobLd — CHANGES_REQUESTED** [review]
Some formatting comments

**Review by BobLd — APPROVED** [review]
Thanks for your help on that!

**Comment by huzhiguan** [comment]
> Thanks for your help on that!

You're welcome. Thank you all for the great project!

**Comment by EliotJones** [comment]
Thank you for the contribution and sorry for the delay in merging.

**Pull request merged**